subxt/src/backend/unstable/mod.rs

```diff
@@ -145,20 +146,30 @@ impl<T: Config> UnstableBackend<T> {
         I: IntoIterator<Item = follow_stream_unpin::BlockRef<T::Hash>> + Send + 'static,
         <I as IntoIterator>::IntoIter: Send,
     {
-        let sub_id = get_subscription_id(&self.follow_handle).await?;
-        let sub_id = Arc::new(sub_id);
         let methods = self.methods.clone();
+        let follow_handle = self.follow_handle.clone();
+
         let headers = self.follow_handle.subscribe().events().flat_map(move |ev| {
-            let sub_id = sub_id.clone();
             let methods = methods.clone();
-
+            let follow_handle = follow_handle.clone();
             let block_refs = f(ev).into_iter();
 
             futures::stream::iter(block_refs).filter_map(move |block_ref| {
-                let sub_id = sub_id.clone();
                 let methods = methods.clone();
+                let follow_handle = follow_handle.clone();
 
                 async move {
+                    // TODO(niklasad1): naive impl which will fetch the subscription ID for every block
```

Comment:
//cc @jsdw suggestions here?

Reply:
The subscription ID is emitted in a Ready message each time the subscription is renewed for whatever reason, but the `.events()` fn ignores it.

So perhaps we could remove the `.events()` above and then keep an eye out for new Ready messages to update the subscription ID here as needed, while manually filtering out the events we don't care about?

We have a similar issue in `unstable/storage_items.rs` where we get a subscription ID once and then subscribe to stuff and prob should do a similar sort of thing there too?